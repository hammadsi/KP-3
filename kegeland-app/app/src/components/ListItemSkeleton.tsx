import React from 'react';
import {StyleSheet, View} from 'react-native';

import Skeleton from './Skeleton';

/**
 * Skeleton component for ListItem.
 * @see {@link Skeleton}
 */
const ListItemSkeleton: React.FC = () => {
  return (
    <View style={styles.skeleton}>
      <Skeleton mode="rounded" width={22} height={22} />
      <Skeleton style={styles.skeletonText} height={22} />
    </View>
  );
};

const styles = StyleSheet.create({
  skeleton: {
    marginVertical: 8,
    padding: 8,
    alignItems: 'center',
    flexDirection: 'row',
  },
  skeletonText: {
    flex: 1,
    marginLeft: 8,
  },
});

export default ListItemSkeleton;
