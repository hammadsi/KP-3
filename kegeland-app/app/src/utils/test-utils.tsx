/* istanbul ignore file */
import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {Provider as PaperProvider} from 'react-native-paper';
import {render, RenderOptions} from '@testing-library/react-native';
import {Provider as ReduxProvider} from 'react-redux';

import {store} from '~state/store';
import {lightTheme} from '~constants/theme';
import {initialStore, mockStore} from '~state/ducks/__mocks__/store.mock';

type TestWrapperProps = {
  children: React.ReactNode;
  store?: typeof store;
};

/**
 * Wrapper component with redux state. Used when testing components
 * which requires redux.
 * @param param0 test wrapper props @see {@link TestWrapperProps}
 * @returns
 */
export const TestWrapper: React.FC<TestWrapperProps> = ({
  children,
  ...props
}) => {
  return (
    <ReduxProvider store={props.store || mockStore(initialStore)}>
      <PaperProvider theme={lightTheme}>
        <NavigationContainer theme={lightTheme}>{children}</NavigationContainer>
      </PaperProvider>
    </ReduxProvider>
  );
};

/**
 * Custom test renderer which wraps components with a redux store provider
 * @param ui the component to render
 * @param options renderer options
 * @returns test rendered componnet
 */
export const customRender = <T extends any>(
  ui: React.ReactElement<T>,
  options?: {
    wrapperProps?: {
      store: any;
    };
    renderOptions?: RenderOptions;
  },
) =>
  render(ui, {
    wrapper: (props) => <TestWrapper {...props} {...options?.wrapperProps} />,
    ...options?.renderOptions,
  });

// re-export everything
export * from '@testing-library/react-native';

// override render method
export {customRender as render};
