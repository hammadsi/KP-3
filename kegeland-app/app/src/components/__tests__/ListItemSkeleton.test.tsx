import React from 'react';

import {render} from '~utils/test-utils';

import ListItemSkeleton from '../ListItemSkeleton';

describe('Test ListItemSkeleton-component', () => {
  it('should render correctly', () => {
    const component = <ListItemSkeleton />;
    const tree = render(component);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
