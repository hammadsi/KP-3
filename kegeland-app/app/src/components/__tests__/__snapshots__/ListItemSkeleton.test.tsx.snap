// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ListItemSkeleton-component should render correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginVertical": 8,
        "padding": 8,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#000000",
          "borderRadius": 11,
          "height": 22,
          "opacity": 0.1,
          "width": 22,
        }
      }
    >
      <View
        style={Object {}}
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#000000",
          "borderRadius": 0,
          "flex": 1,
          "height": 22,
          "marginLeft": 8,
          "opacity": 0.1,
          "width": undefined,
        }
      }
    >
      <View
        style={Object {}}
      />
    </View>
  </View>
</View>
`;
